addons/event/models/event_registration.py

```diff
@@ -93,18 +104,17 @@ def _compute_date_closed(self):
                 else:
                     registration.date_closed = False
 
-    @api.constrains('event_id', 'state')
-    def _check_seats_limit(self):
-        for registration in self:
-            event_id = registration.event_id
-            if event_id.seats_limited and event_id.seats_max and event_id.seats_available < (1 if registration.state == 'draft' and event_id.auto_confirm else 0):
-                raise ValidationError(_('There are no more available seats for %s. Raise the limit or remove some other confirmed registrations first.', event_id.name))
+    @api.constrains('event_id', 'event_ticket_id', 'state')
+    def _check_seats_availability(self):
+        """Checks that neither the event nor the ticket have more confirmed
+        seats than allowed by their respective seats_max.
 
-    @api.constrains('event_ticket_id', 'state')
-    def _check_ticket_seats_limit(self):
-        for record in self:
-            if record.event_ticket_id.seats_max and record.event_ticket_id.seats_available < 0:
-                raise ValidationError(_('No more available seats for this ticket'))
+        :param self: Registration records requiring available seats an event
+            (ticket) (states: "open", "done" and "draft" if event `auto_confirm`)
+        """
+        for record in itertools.chain(set(self.mapped("event_id")),
+                                      set(self.mapped("event_ticket_id"))):
+            record ._check_seats_availability()
```

Comment:
First, `mapped` is automatic on relational fields.
Second, it already deduplicates its result, so the `set()` is useless.
Third, mixing records from different models in `itertools.chain()` is :hankey: 
Fourth, `_check_seats_availability()` works on recordsets, so the `for` loop is simply useless.
```suggestion
        self.event_id._check_seats_availability()
        self.event_ticket_id._check_seats_availability()
```